products.js

```diff
@@ -0,0 +1,308 @@
+//     }
+//
+//     function tagHasGoodCss(fnode){
+//       if(fnode.element.id.match(/price/i) || fnode.element.classList.contains(/price/i)){
+//         return 2;
+//       }
+//       return 1;
+//     }
+//
+//     function notSavingsAmount(fnode){
+//       if(fnode.element.childNodes[0] && fnode.element.childNodes[0].nodeValue &&
+//         (fnode.element.childNodes[0].nodeValue.includes('-') || window.getComputedStyle(fnode.element).getPropertyValue('text-decoration') === 'line-through')){
+//         return 0;
+//       }
+//       return 1;
+//     }
+//
+//
+//     const rules = ruleset(
+//       //get all elements that could contain the price
+//       rule(dom('span, div, li'), type('priceish')),
+//
+//       //bonus if direct text (not children) contains a dollar sign
+//       rule(type('priceish'), score(hasDollarSign)),
+//
+//       //look for good css within tag
+//       rule(type('priceish'), score(tagHasGoodCss)),
+//
+//       //not to be confused with amount off (minus sign, crossed off)
+//       rule(type('priceish'), score(notSavingsAmount)),
+//
+//       //return image with max score
+//       rule(type('priceish').max(), out('product-price'))
+//
+//     );
+//
+//     function contentFnodes(doc) {
+//         return rules.against(doc).get('product-price');
+//     }
+//
+//     return contentFnodes;
+// }
+
+/**
+ * Maintain state as we compare a series of DOMs, reporting the percent
+ * difference at the end.
+ */
+class DiffStats {
+    constructor(contentFnodes, feature) {
+        this.lengthOfExpectedTexts = 0;
+        this.lengthOfDiffs = 0;
+        this.feature = feature;
+        if (feature === 'image') {
+          this.contentFnodes = contentFnodes || tunedImageFnodes();
+        } else if (feature === 'title'){
+          this.contentFnodes = contentFnodes || tunedTitleFnodes();
+        } else {
+          this.contentFnodes = contentFnodes || tunedPriceFnodes();
+        }
+    }
+
+    compare(expectedDom, sourceDom) {
+        var expectedText;
+        var gotText;
+        if (this.feature === 'image') {
+          //compare images by src
+          expectedText = expectedDom.body.firstChild.src;
+          gotText = this.contentFnodes(sourceDom).map(fnode => fnode.element.src)[0];
+        } else if (this.feature === 'title') {
+          //compare innerHTML text of titles
+          expectedText = expectedDom.head.firstChild.innerHTML;
+          gotText = this.contentFnodes(sourceDom).map(fnode => fnode.element.innerHTML)[0];
+        } else if (this.feature === 'price') {
+          //strip whitespace and dollar sign if there is one when comparing price
+          expectedText = expectedDom.body.firstChild.textContent.replace('$', '').trim();
+          gotText = this.contentFnodes(sourceDom).map(fnode => fnode.element.textContent.replace('$', '').trim())[0];
+        }
+
+        this.lengthOfExpectedTexts++;
+        if(expectedText !== gotText) {
+          this.lengthOfDiffs++;
```

Comment:
`lengthOfDiffs` is no longer an accurate name either. Maybe something along the lines of "cost" or "deviation"?